File: src/main/java/top/dsbbs2/bukkitcord/api/IPlatform.java
package top.dsbbs2.bukkitcord.api;

import java.util.*;

public interface IPlatform {
    IPluginManager getPluginManager();
    void dispatchCommand(ICommandSender sender, String cmd);
    ICommandSender getConSoleSender();
    IPlayer getPlayer(String n);
    IPlayer getPlayer(UUID u);
    void broadcast(String string);
    void broadcast(String string, String permission);
    ArrayList<IPlayer> getOnlinePlayers();
    ArrayList<String> getOnlinePlayersName();
    boolean isBungeeCord();
}

File: src/main/java/top/dsbbs2/bukkitcord/api/IPlatform.kt
package top.dsbbs2.bukkitcord.api

import java.util.*

interface IPlatform {
    val pluginManager: IPluginManager?
    fun dispatchCommand(sender: ICommandSender?, cmd: String?)
    val conSoleSender: ICommandSender?
    fun getPlayer(n: String?): IPlayer?
    fun getPlayer(u: UUID?): IPlayer?
    fun broadcast(string: String?)
    fun broadcast(string: String?, permission: String?)
    val onlinePlayers: ArrayList<IPlayer?>?
    val onlinePlayersName: ArrayList<String?>?
    val bungeeCord: Boolean
}